zk/txpool/pool_zk_limbo.go

```diff
@@ -333,10 +385,12 @@ func (p *TxPool) finalizeLimboOnNewBlock(limboTxs *types.TxSlots) {
 
 // should be called from within a locked context from the pool
 func (p *TxPool) isTxKnownToLimbo(hash common.Hash) bool {
-	for _, limbo := range p.limbo.limboBatches {
-		for _, limboTx := range limbo.Transactions {
-			if limboTx.Hash == hash {
-				return true
+	for _, limboBatch := range p.limbo.limboBatches {
```

Comment:
the comment above suggests this should be done under a lock?

Reply:
It is. This function is invoked from trimLimboSlots and trimLimboSlots is invoked from OnNewBlock which locks the pool